packages/core/content-type-builder/admin/src/components/IconPicker/index.js

```diff
@@ -0,0 +1,146 @@
+import React, { useState, useRef } from 'react';
+import { Box, Flex, Icon, Typography, Searchbar, IconButton } from '@strapi/design-system';
+import * as Icons from '@strapi/icons';
+import { useIntl } from 'react-intl';
+import { inputFocusStyle } from '@strapi/design-system';
+import styled from 'styled-components';
+import { getTrad } from '../../utils';
+
+const IconPickerWrapper = styled(Flex)`
+  position: relative;
+
+  label {
+    border-radius: 4px;
+    ${inputFocusStyle}
+  }
+
+  input {
+    position: absolute;
+    opacity: 0;
+  }
+`;
+
+const IconPick = ({ iconKey, name, onChange, isSelected }) => {
+  return (
+    <label htmlFor={iconKey}>
+      <input
+        id={iconKey}
+        name={name}
+        checked={isSelected}
+        onChange={onChange}
+        value={iconKey}
+        type="radio"
+      />
+      <Box padding={2} cursor="pointer" hasRadius background={isSelected ? 'primary200' : null}>
+        <Icon as={Icons[iconKey]} color={isSelected ? 'primary600' : 'neutral300'} />
```

Comment:
Word of caution: when we removed icons we did not migrate the database and let the old keys in there. That means you might receive a string here, referencing an old icon.